admin/aws.py

```diff
@@ -290,7 +290,7 @@ def perform_upload_s3_key_recursively(dispatcher, intent):
                 UploadToS3(
                     source_path=intent.source_path,
                     target_bucket=intent.target_bucket,
-                    target_key=intent.target_key,
+                    target_key="%s/%s" % (intent.target_key, file),
```

Comment:
IIRC I didn't do this (or something similar) because AWS key names may end with `/`. Can you confirm that this isn't the case or work around it.